### ISSUE TITLE
Resources file in Business layer

### ISSUE THREAD

**Issue by Chicagoan2016** [MarimerLLC/cslaforum#660]
Hi All,
following the example of @jonnybee  in RulesTutorial, I have created a Resources File in Business Layer to store 'hard coded' messages for business rules. I have also started to use the Resources file for storing default values of my 'string' properties. Does the resources file gets serialized/Deserialized by .NET or do we have to do it ourselves?

Kind Regards

**Version and Platform**
CSLA version: 4.7.100
OS: Windows, Linux, iOS, Android, etc.
Platform: WinForms, WPF, ASP.NET Core, MVC, Xamarin, etc.

**Comment by rockfordlhotka** [comment]
Resource files aren't serialized, they are part of your build output and are deployed to your devices along with the core DLL you deploy.

**Comment by Chicagoan2016** [comment]
Thank you @rockfordlhotka , I am assuming  it is safe to use Resources file with Csla ?? for 
1) hard coded business rules messages.
2) Default value of string properties (Csla properties)

kind regards

**Comment by jonnybee** [comment]
Yes, as far as CSLA is concerned a string variable in memory is just that, whether it comes from user input or is read from a resource file or another source.

**Comment by Chicagoan2016** [comment]
Thank you @jonnybee , In #648 , I mentioned storing hard coded strings in 'const' strings in static classes
but I wasn't sure if those values will be available to Csla mobile objects.  So I move them to Resource files.
I wonder what fellow Csla coders are using for hard coded strings especially for default values of string properties?

Kind Regards

**Comment by jonnybee** [comment]
Use resources if you need to support several languages. 

A string is just a string and for CSLA it doesn't matter if the default value came from a const or a member variable or a resource file or the PropertyInfo.Default. A CSLA mobile object will serialize it's properties just fine .

**Comment by Chicagoan2016** [comment]
Thank you @jonnybee , This 

> A string is just a string and for CSLA it doesn't matter if the default value came from a const or a member variable or a resource file or the PropertyInfo.Default. A CSLA mobile object will serialize it's properties just fine 

> .

clears everything, I was worried that if I used const strings in a static class for default values,the default values might not be there when Csla objects move between client and server.
